### PR TITLE
[Fix] Convert SyncBN to BN when training on DP

### PR DESCRIPTION
**Pull request by sennnnn**
### Incompatible between DP and SyncBN

run this command without setting `--launcher`:

```base
python tools/train.py [config_path]
```

The python environment will report errors about process group: 

```None
RuntimeError: Default process group has not been initialized, please make sure to call init_process_group.
```

This error is caused by `SyncBN`. When `SyncBN.training` is True, `SyncBN` need to init `process_group`. However, `process_group` is only valid when `DDP`.

The situation that SyncBN is valid:
* `process_group` is not None or `torch.distributed.group.WORLD` is not `None`;
* `SyncBN.training` is False or `SyncBN.eval()`;

Based on those mentioned above, we convert SyncBN to BN when training on DP.

This PR is blocked by MMCV [#1253](https://github.com/open-mmlab/mmcv/pull/1253)

**MMCV PR #1253 may be released in MMCV 1.3.13 version, the mmcv compatibility of mmseg 0.18 will larger than 1.3.13.**

**The CI will pass when MMCV 1.3.13 version release**

### CODE REVIEW

**Comment by xvjiarui** [comment]
May move to `models/utils`

File: tools/train.py [hunk 1]
```diff
@@ -133,6 +135,14 @@ def main():
         test_cfg=cfg.get('test_cfg'))
     model.init_weights()
 
+    # SyncBN is not support for single gpu
```
Comment (Junjun2016) [inline tools/train.py:138]:
```suggestion
    # SyncBN is not support for DP
```

File: tools/train.py [hunk 2]
```diff
@@ -133,6 +135,14 @@ def main():
...
+    if not distributed:
+        warnings.warn(
+            'SyncBN only support DDP. In order to compat with DP, we convert '
+            'SyncBN tp BN. Please to use dist_train.py which has official '
```
Comment (Junjun2016) [inline tools/train.py:142]:
```suggestion
            'SyncBN to BN. Please use dist_train.sh which can '
```

File: tools/train.py [hunk 3]
```diff
@@ -133,6 +135,14 @@ def main():
...
+            'support to avoid this problem.')
```
Comment (Junjun2016) [inline tools/train.py:143]:
```suggestion
            'this problem.')
```

**Review by Junjun2016 — CHANGES_REQUESTED** [review]
Please fix the conflict.

**Review by Junjun2016 — APPROVED** [review]

**Comment by xvjiarui** [comment]
https://github.com/open-mmlab/mmcv/pull/1253

**Comment by Junjun2016** [comment]
We can import from mmcv after it merged.

**Comment by Junjun2016** [comment]
Can import `revert_sync_batchnorm` from https://github.com/open-mmlab/mmcv/pull/1253

File: docker/serve/Dockerfile [hunk 4]
```diff
@@ -3,7 +3,7 @@ ARG CUDA="10.1"
 ARG CUDNN="7"
 FROM pytorch/pytorch:${PYTORCH}-cuda${CUDA}-cudnn${CUDNN}-devel
 
-ARG MMCV="1.3.12"
+ARG MMCV="1.3.13"
```
Comment (Junjun2016) [inline docker/serve/Dockerfile:6]: Should also update mmcv version [here](https://github.com/open-mmlab/mmsegmentation/blob/master/docker/Dockerfile#L4).

File: tools/train.py [hunk 5]
```diff
@@ -134,6 +136,14 @@ def main():
         test_cfg=cfg.get('test_cfg'))
     model.init_weights()
 
+    # SyncBN is not support for DP
+    if not distributed:
+        warnings.warn(
+            'SyncBN only support DDP. In order to compat with DP, we convert '
```
Comment (Junjun2016) [inline tools/train.py:142]:
```suggestion
            'SyncBN is only supported with DDP. To be compatible with DP, we convert '
```

**Review by Junjun2016 — COMMENTED** [review]
Please fix the conflict.

**Review by Junjun2016 — APPROVED** [review]
LGTM

**Comment by xvjiarui** [comment]
Please upgrade mmcv requirement.

**Review by Junjun2016 — COMMENTED** [review]
https://github.com/open-mmlab/mmsegmentation/blob/master/docs/get_started.md#installation

Update mmcv dependence of master branch.

**Review by Junjun2016 — APPROVED** [review]
LGTM

**Pull request merged**